File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        HashMap<String,Integer> map = new HashMap<>();
        
        for(String key : participant) {
            map.put(key, map.getOrDefault(key,0) + 1);
            // System.out.println(map.get(key));
        }
        
        for(String complete : completion) {
            if(map.containsKey(complete)) {
                map.put(complete,map.getOrDefault(complete,0) - 1);
                // System.out.println(map.get(complete));
            }
        }
        
        for(String key : map.keySet()) {
            if(map.get(key).equals(0)) {
            }
            else {
                answer = key;
            }
        }
        
        return answer;
        
    }
}

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    
    data = dict()
    
    for name in participant:
        data.setdefault(name,0)
        data[name] += 1

    for complete in completion:
        if complete in data.keys():
            data[complete] -= 1
    
    answer = ''      
    for key,value in data.items():
        if (value != 0):
            answer = key
        
    return answer